FinniversKit/Sources/Components/MonthAndYearPickerView/MonthAndYearPickerView.swift

```diff
@@ -0,0 +1,190 @@
+        calendar.timeZone = timeZone
+
+        translatesAutoresizingMaskIntoConstraints = false
+
+        addSubview(pickerView)
+        pickerView.fillInSuperview()
+    }
+
+    public override var intrinsicContentSize: CGSize {
+        pickerView.systemLayoutSizeFitting(
+            CGSize(width: frame.width, height: UIView.layoutFittingCompressedSize.height),
+            withHorizontalFittingPriority: .required,
+            verticalFittingPriority: .fittingSizeLevel
+        )
+    }
+
+    private func selectedDateIsBefore(_ minimumDate: Date) -> Bool {
+        calendar.compare(minimumDate, to: selectedDate, toGranularity: .month) == .orderedDescending
+    }
+
+    private func selectedDateIsAfter(_ maximumDate: Date) -> Bool {
+        calendar.compare(selectedDate, to: maximumDate, toGranularity: .month) == .orderedDescending
+    }
+
+    public func setSelectedDate(_ date: Date, animated: Bool) {
+        selectedDate = date
+        setPickerViewSelection(date, animated: animated)
+    }
+
+    private func setPickerViewSelection(_ date: Date, animated: Bool) {
+        let components = calendar.dateComponents([.month, .year], from: date)
+
+        guard
+            let monthComponent = components.month,
+            let yearComponent = components.year,
+            let yearIndex = years.firstIndex(of: yearComponent)
+        else {
+            return
+        }
+
+        let monthIndex = monthComponent - 1
+        pickerView.selectRow(monthIndex, inComponent: .month, animated: animated)
+        pickerView.selectRow(yearIndex, inComponent: .year, animated: animated)
+    }
+}
+
+// MARK: - UIPickerViewDataSource
+extension MonthAndYearPickerView: UIPickerViewDataSource {
+    public func numberOfComponents(in pickerView: UIPickerView) -> Int {
+        UIPickerView.Component.allCases.count
+    }
+
+    public func pickerView(_ pickerView: UIPickerView, numberOfRowsInComponent component: Int) -> Int {
+        guard let component = UIPickerView.Component(rawValue: component) else {
+            return 0
+        }
+
+        switch component {
+        case .month:
+            return months.count
+        case .year:
+            return years.count
+        }
+    }
+}
+
+// MARK: - UIPickerViewDelegate
+extension MonthAndYearPickerView: UIPickerViewDelegate {
+    public func pickerView(_ pickerView: UIPickerView, titleForRow row: Int, forComponent component: Int) -> String? {
+        guard let component = UIPickerView.Component(rawValue: component) else {
+            return nil
+        }
+
+        switch component {
+        case .month:
+            return months[row]
+        case .year:
+            return String(years[row])
+        }
+    }
+
+    public func pickerView(_ pickerView: UIPickerView, didSelectRow row: Int, inComponent component: Int) {
+        let month = pickerView.selectedRow(inComponent: .month) + 1
+        let year = years[pickerView.selectedRow(inComponent: .year)]
```

Comment:
What if we use the `safe` subscript here? 

```suggestion
        guard let year = years[safe: pickerView.selectedRow(inComponent: .year)] else { return }
```